tests_new/integration_tests/modules/worksheets/test_worksheet.py

```diff
@@ -59,3 +63,83 @@ def test_update_worksheet_unauthorized(client2, worksheet1):
     assert_that(update_worksheet).raises(GqlError).when_called_with(
         client2, worksheet1.worksheetUri, worksheet1.label, 'updated desc', worksheet1.tags
     ).contains('UnauthorizedOperation', 'UPDATE_WORKSHEET')
+
+
+def test_run_athena_sql_query(client1, worksheet1, persistent_env1):
+    sql_query = 'SHOW DATABASES;'
+    rows = run_athena_sql_query(
+        client=client1,
+        query=sql_query,
+        environment_uri=persistent_env1.environmentUri,
+        worksheet_uri=worksheet1.worksheetUri,
+    ).rows
+    assert_that(rows).is_not_empty()
+    db_names = [r.cells[0].value for r in rows]
+    assert_that(db_names).contains('default')
+
+
+def test_run_athena_sql_query_unauthorized(client2, worksheet1, persistent_env1):
+    sql_query = 'SHOW DATABASES;'
+    assert_that(run_athena_sql_query).raises(GqlError).when_called_with(
+        client2, sql_query, persistent_env1.environmentUri, worksheet1.worksheetUri
+    ).contains('UnauthorizedOperation', 'RUN_ATHENA_QUERY')
+
+
+@pytest.mark.skipif(not config.get_property('modules.worksheets.features.nlq'), reason='Feature Disabled by Config')
+def test_text_to_sql(client1, worksheet1, persistent_env1):
+    prompt = 'Write me a query to list the databases I have access to in Athena'
+    response = text_to_sql(
+        client=client1,
+        prompt=prompt,
+        environment_uri=persistent_env1.environmentUri,
+        worksheet_uri=worksheet1.worksheetUri,
+        database_name='',
+        table_names=[],
+    )
+    # Results are nondeterministic - just asserting the response is not None
+    assert_that(response).is_not_none()
+
+
+@pytest.mark.skipif(not config.get_property('modules.worksheets.features.nlq'), reason='Feature Disabled by Config')
+def test_text_to_sql_unauthorized(client2, worksheet1, persistent_env1):
+    prompt = 'Write a query to access data in athena'
+    assert_that(text_to_sql).raises(GqlError).when_called_with(
+        client2, prompt, persistent_env1.environmentUri, worksheet1.worksheetUri, ''
+    ).contains('UnauthorizedOperation', 'RUN_ATHENA_QUERY')
+
+
```

Comment:
Commented code. I am not sure what we should do with this code block

Reply:
left it here to hear others opinions - not sure what is best approach is, will think on this and get back tomorrow with some thoughts